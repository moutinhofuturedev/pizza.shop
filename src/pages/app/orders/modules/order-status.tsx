export enum OrderStatusType {
  Pending = 'pending',
  Canceled = 'canceled',
  Processing = 'processing',
  Delivering = 'delivering',
  Delivered = 'delivered',
}

interface OrderStatusProps {
  status: OrderStatusType
}

const OrderStatusMap: Record<OrderStatusType, string> = {
  [OrderStatusType.Pending]: 'Pendente',
  [OrderStatusType.Canceled]: 'Cancelado',
  [OrderStatusType.Processing]: 'Em preparo',
  [OrderStatusType.Delivering]: 'Em entrega',
  [OrderStatusType.Delivered]: 'Entregue',
}

const StatusColorMap: Record<OrderStatusType, string> = {
  [OrderStatusType.Pending]: 'bg-blue-500',
  [OrderStatusType.Canceled]: 'bg-red-500',
  [OrderStatusType.Processing]: 'bg-yellow-500',
  [OrderStatusType.Delivering]: 'bg-orange-500',
  [OrderStatusType.Delivered]: 'bg-emerald-500',
}

export const OrderStatus = ({ status }: OrderStatusProps) => {
  return (
    <div className="flex items-center gap-2">
      <span className={`h-2 w-2 rounded-full ${StatusColorMap[status]}`} />
      <span className="font-medium text-muted-foreground">
        {OrderStatusMap[status]}
      </span>
    </div>
  )
}
