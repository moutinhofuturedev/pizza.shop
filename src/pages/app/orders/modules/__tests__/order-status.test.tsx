import { render } from '@testing-library/react'

import { OrderStatus, OrderStatusType } from '../order-status'

describe('<OrderStatus />', () => {
  it('should render display order status correctly with props', () => {
    const wrapper = render(<OrderStatus status={OrderStatusType.Delivered} />)

    const statusText = wrapper.getByText('Entregue')

    expect(statusText).toBeInTheDocument()
  })

  it('should render display order status correctly with class badge color', () => {
    const wrapper = render(<OrderStatus status={OrderStatusType.Canceled} />)

    const statusText = wrapper.getByText('Cancelado')
    const orderStatus = wrapper.getByTestId('order-status')

    expect(statusText).toBeInTheDocument()
    expect(orderStatus).toHaveClass('bg-red-500')
  })

  it('should render display order status correctly', () => {
    const { container } = render(
      <OrderStatus status={OrderStatusType.Pending} />,
    )

    expect(container).toMatchSnapshot()
  })
})
