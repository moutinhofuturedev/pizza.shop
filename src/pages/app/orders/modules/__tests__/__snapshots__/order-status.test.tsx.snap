// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<OrderStatus /> > should render display order status correctly 1`] = `
<div>
  <div
    class="flex items-center gap-2"
  >
    <span
      class="h-2 w-2 rounded-full bg-blue-500"
      data-testid="order-status"
    />
    <span
      class="font-medium text-muted-foreground"
    >
      Pendente
    </span>
  </div>
</div>
`;
